ocs_ci/utility/utils.py

```diff
@@ -366,25 +373,33 @@ def run_cmd(cmd, **kwargs):
         (str) Decoded stdout of command
 
     """
-    log.info(f"Executing command: {cmd}")
-    if isinstance(cmd, str):
-        cmd = shlex.split(cmd)
-    r = subprocess.run(
-        cmd,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        stdin=subprocess.PIPE,
-        **kwargs
-    )
-    log.debug(f"Command output: {r.stdout.decode()}")
-    if r.stderr and not r.returncode:
-        log.warning(f"Command warning:: {r.stderr.decode()}")
-    if r.returncode:
-        raise CommandFailed(
-            f"Error during execution of command: {cmd}."
-            f"\nError is {r.stderr.decode()}"
+    try:
+        quiet_state = quiet_logs(quiet=kwargs.pop('quiet', False))
+        log.info(f"Executing command: {cmd}")
+        if isinstance(cmd, str):
+            cmd = shlex.split(cmd)
+        r = subprocess.run(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            stdin=subprocess.PIPE,
+            **kwargs
         )
-    return r.stdout.decode()
+        log.debug(f"Command output: {r.stdout.decode()}")
+        if r.stderr and not r.returncode:
+            log.warning(f"Command warning:: {r.stderr.decode()}")
+        logging.disable(logging.NOTSET)
+        if r.returncode:
+            raise CommandFailed(
+                f"Error during execution of command: {'quiet command' if quiet_state else cmd}."
+                f"\nError is {r.stderr.decode()}"
+            )
+        return r.stdout.decode()
+    except Exception as e:
```

Comment:
3. This will catch the CommandFailed in line 393 so needs to move earlier - see comment 2
4. No need for except - just finally is enough

Reply:
Regarding 3 - I commented on the respective comment.
Regarding 4 - wouldn't we like to log the failure?

Reply:
3. ack
4. yes we do